clang/lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -1480,6 +1480,14 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
       return false;
     break;
   }
+  case Type::PredefinedSugar: {
+    const auto *TP1 = cast<PredefinedSugarType>(T1);
+    const auto *TP2 = cast<PredefinedSugarType>(T2);
+
+    if (TP1->getKind() != TP2->getKind())
+      return false;
+    break;
```

Comment:
If you have an underlying type, then this needs to do a structural comparison of the underlying types as well.

If you have just a fixed list with one type per kind, then just comparing the kinds is fine.